src/core/main.c

```diff
@@ -1590,6 +1582,16 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
+        /* Initialize default unit */
+        if (!arg_default_unit) {
+                r = free_and_strdup(&arg_default_unit, SPECIAL_DEFAULT_TARGET);
```

Comment:
why bother with free_and_strdup() we only call this if arg_default_unit is NULL anyway?

This should really be simplified to:

```c
arg_default_unit = strdup(SPECIAL_DEFAULT_TARGET);
if (!arg_default_unit) {
        r = log_oom(); 
        …
}
```